main.go

```diff
@@ -55,18 +55,35 @@ func main() {
 	}
 	s3Cli := s3.NewFromConfig(cfg)
 
-	// List objects matching bucket / prefix
-	res, err := s3Cli.ListObjects(ctx, &s3.ListObjectsInput{
-		Bucket: aws.String(opts.Bucket),
-		Prefix: aws.String(opts.Prefix),
-	})
-	if err != nil {
-		log.Fatal().Err(err).Msg("Unable to list objects")
+	maxKeys := int32(1000)
+	var elems s3.ListObjectsOutput
+
+	p := s3.NewListObjectsV2Paginator(
+		s3Cli,
+		&s3.ListObjectsV2Input{
+			Bucket:  aws.String(opts.Bucket),
+			Prefix:  aws.String(opts.Prefix),
+			MaxKeys: &maxKeys,
+		},
+		func(o *s3.ListObjectsV2PaginatorOptions) {
+			if v := int32(maxKeys); v != 0 {
+				o.Limit = v
+			}
+		})
```

Comment:
Since 1000 keys is the default I think you could simply do as shown [here](https://aws.github.io/aws-sdk-go-v2/docs/migrating/#example-7) : 

```suggestion
	p := s3.NewListObjectsV2Paginator(s3Cli,&s3.ListObjectsV2Input{
		Bucket:  aws.String(opts.Bucket),
		Prefix:  aws.String(opts.Prefix),
	})
```

The `func(o *s3.ListObjectsV2PaginatorOptions) { ... }` is a pattern to set options in go (but usually there are helpers and you do not write the func yourself, but not here). 
Looking into [ListObjectsV2Input](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#ListObjectsV2Input) and [ListObjectsV2PaginatorOptions](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#ListObjectsV2PaginatorOptions) it seems that both MaxKeys and Limit have 1000 as default.